backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/util/lang/impl/TableRecordReference.java

```diff
@@ -211,13 +211,21 @@ public static TableRecordReference of(@JsonProperty("tableName") final String ta
 	}
 
 	@Nullable
-	public static TableRecordReference ofNullable(@Nullable String tableName, final int recordId)
+	public static TableRecordReference ofNullable(final @Nullable String tableName, final int recordId)
 	{
 		return tableName != null && !Check.isBlank(tableName) && recordId >= 0
 				? of(tableName, recordId)
 				: null;
 	}
 
+	@Nullable
+	public static TableRecordReference ofNullable(final @Nullable String tableName, @Nullable final RepoIdAware recordId)
+	{
+		return tableName != null && !Check.isBlank(tableName) && recordId != null && recordId.getRepoId() >= 0
```

Comment:
IMHO you don't have to check recordId.getRepoId() >= 0.
If you have a RepoIdAware instance, you can assume it's valid.

Reply:
done